### ISSUE TITLE
Kotlin: @Transient on property with custom setter and getter not working

### ISSUE THREAD

**Issue by saeednt** [objectbox/objectbox-java#613]
I annotated one of the properties of my class with @Transient  but it has setter and getter as follows:
```
@Transient
var date: LocalDate = LocalDate.MIN
        get() = LocalDate.of(year, month, day)
        set(value) {
            field = value
            year = value.year
            month = value.monthValue
            day = value.dayOfMonth
        }
```
ObjectBox throws unsupported field error. This is the error:

> 
> [ObjectBox] Field type "org.threeten.bp.LocalDate" is not supported. Consider making the target an @Entity, or using @Convert or @Transient on the field (see docs). (com.rashin.reminder.entity.Event.date)
>     private org.threeten.bp.LocalDate date

**Comment by greenrobot-team** [comment]
Huh, I'm a little at a loss here. Looking at the decompiled byte-code the annotation is simply not added.

Using `@field:Transient` or Kotlin's `@Transient` also does not work.

If either getter or setter is removed, the annotation is added properly. So I believe this has something to do with how Kotlin handles properties.

If only the ObjectBox' `@Transient` annotation is added, Kotlin will create a synthetic method which has the annotations. But it's marked as deprecated:
```
   /** @deprecated */
   // $FF: synthetic method
   @Transient
   public static void date$annotations() {
   }
```

Not sure what we should do here.
-ut

**Comment by greenrobot-team** [comment]
OK, turns out this property is not considered to have a backing field because it has a custom getter *and* setter. Hence the annotation will not work.

Moshi has the same issue: https://github.com/square/moshi/issues/609
-ut

**Comment by greenrobot-team** [comment]
@saeednt Note that your example is broken. You re-create a `LocalDate` any time you access the `date` property instead of using the set value. I suggest doing something like:
```kotlin
@Transient
var date: LocalDate = LocalDate.of(year, month, day)
    set(value) {
        field = value
        year = value.year
        month = value.monthValue
        day = value.dayOfMonth
    }
```
This also won't have your reported issue.
-ut

**Comment by cdongieux** [comment]
@greenrobot-team I don't known if it can help, but it worked fine with Kotlin 1.2.71. I saw this bug when upgrading to Kotlin 1.3.10.

Regards.

**Comment by abelkov** [comment]
This should be fixed in the upcoming Kotlin version 1.3.30, see https://youtrack.jetbrains.com/issue/KT-29507.

**Comment by greenrobot-team** [comment]
This appears to have been fixed. Can no longer verify using Kotlin 1.3.41.

Using either the Kotlin or the ObjectBox `@Transient` annotation works.

Closing this then.
